packages/nuxt3/src/core/builder.ts

```diff
@@ -48,7 +48,7 @@ function watch (nuxt: Nuxt) {
     ]
   })
 
-  const watchHook = debounce(debounce.promise((event: 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir', path: string) => nuxt.callHook('builder:watch', event, path)), 1)
+  const watchHook = debounce((event: 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir', path: string) => nuxt.callHook('builder:watch', event, path), 25, { leading: true })
```

Comment:
This is also not a frequent operation. Adding debounce timeout help to avoid rebuild on bulk fs actions.

```suggestion
  const watchHook = debounce((event: 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir', path: string) => nuxt.callHook('builder:watch', event, path), 100)
```